app/javascript/src/components/editor/CodeMirror.svelte

```diff
@@ -195,12 +195,13 @@
         if (shiftKey) {
           if (!/^\s/.test(insert[0]) && !(insert.includes("\n ") || insert.includes("\n\t"))) return { range: EditorSelection.range(from, to) }
 
-          insert = insert.replaceAll(/\n[ \t]/g, "\n").substring(insert.search(/\S/) ? 1 : 0, insert.length)
+          insert = (/^\n/.test(insert[0]) ? "\n" : "").concat(insert.replaceAll(/\n[ \t]/g, "\n").substring(insert.search(/\S/) ? 1 : 0, insert.length))
```

Comment:
This is a bit much to take in, maybe we can separate it to two lines

```suggestion
          insert = insert.replaceAll(/\n[ \t]/g, "\n").substring(insert.search(/\S/) ? 1 : 0, insert.length)
          insert = (/^\n/.test(insert[0]) ? "\n" : "").concat(insert)
```

Reply:
yeah i agree, but not sure if thats going to work. if you make it two lines its going to `test(insert[0])` after insert has already had its linebreaks replaced right?

Reply:
Hmm true, might need to write it to a new variable first, I'm not sure what the best solution would be

Reply:
alright i made a new variable first. bit more code but its probably worth it 😃